src/test/scala/ai/privado/languageEngine/java/DummyJavaTest.scala

```diff
@@ -1,33 +1,36 @@
 package ai.privado.languageEngine.java
 
 import ai.privado.model.Constants
+import ai.privado.testfixtures.JavaFrontendTestSuite
 import io.shiftleft.semanticcpg.language.*
 import io.circe.Json
 import io.circe.syntax.EncoderOps
 
-class DummyJavaTest extends JavaTestCpgBase {
+class DummyJavaTest extends JavaFrontendTestSuite {
```

Comment:
Questions
- How to use `custom rule` for a test case?
- Similarly how to use `PrivadoInput`?
- If we intend to write any test case on dataFlowCache, auditCache, s3DatabaseDetailsCache, appCache propertyFilterCache How we can achieve that?

The reason behind not following Joern's style of writing source code directly is
- In privado-core the content of sample source code used is more than that of joern and spanning across multiple files, leading to the source code occupying a lot of the test case space
- If we use a directory, similar code, which we often end up repeating, can be placed at one


Regarding 
```
Also in earlier case for each
`in` clause will execute the scan every time for sub sections of the
unit test. With the updated implementation, now the scan will be invoked
only once for each `should` section.
```

The 3 different styles of invoking the frontend `withCpg`, `withJson`, `withCpgAndJson` are proposed to fit the different needs of writing a test case.
Example - sometimes we would write test case only on the cpg, sometimes only on outputJson, and sometimes will need both.

The invocation inside `in` was done to show the different usages, the same can be achieved by using `withCpgAndOutputJson`, it that seems inefficient, we can introduce a similar `generateScanResult` on top of `CPG` to achieve what has been proposed

Reply:
> Questions
> 
> * How to use `custom rule` for a test case?
> * Similarly how to use `PrivadoInput`?
> * If we intend to write any test case on dataFlowCache, auditCache, s3DatabaseDetailsCache, appCache propertyFilterCache How we can achieve that?

```
    val privadoInput = PrivadoInput()
    val cpg = code("""..
        |""".stripMargin).withPrivadoInput(privadoInput).generateScanResult()
```
there are similar `withRuleCache`, `withAuditCache` etc implementations to pass on respective input while creating the CPG in the unit test itself.  

> The reason behind not following Joern's style of writing source code directly is
> 
> * In privado-core the content of sample source code used is more than that of joern and spanning across multiple files, leading to the source code occupying a lot of the test case space

With sample code in other places, unit tests no longer remain readable. I think even if the unit test code becomes more in a file that should be acceptable as long as it remains more readble. A probable way out to reduce the size of the file is to limit one combination of samples per unit test file.

> * If we use a directory, similar code, which we often end up repeating, can be placed at one

If there are more combinations of unit tests to be written for the same code sample combination. One should write those additional unit tests inside the same `should` section with more `in` sections. 

> Regarding
> 
> ```
> Also in earlier case for each
> `in` clause will execute the scan every time for sub sections of the
> unit test. With the updated implementation, now the scan will be invoked
> only once for each `should` section.
> ```
> 
> The 3 different styles of invoking the frontend `withCpg`, `withJson`, `withCpgAndJson` are proposed to fit the different needs of writing a test case. Example - sometimes we would write test case only on the cpg, sometimes only on outputJson, and sometimes will need both.
> 
> The invocation inside `in` was done to show the different usages, the same can be achieved by using `withCpgAndJson`, it that seems inefficient, we can introduce a similar `generateScanResult` on top of `CPG` to achieve what has been proposed

Yea that's perfectly understood, but devs will follow the way given in the sample code. They might not think over the optimisations while implementing.